core/trino-main/src/main/java/io/trino/sql/gen/AndCodeGenerator.java

```diff
@@ -58,7 +57,7 @@ public BytecodeNode generateExpression(BytecodeGeneratorContext generator)
             RowExpression term = terms.get(i);
             block.append(generator.generate(term));
 
-            IfStatement ifWasNull = new IfStatement(format("if term %s wasNull...", i))
+            IfStatement ifWasNull = new IfStatement("if term " + i + " wasNull...")
```

Comment:
These make the code less readable, IMO. Also, the code generation is not such a frequent operation that performance of concatenation vs String.format is a concern.

Reply:
I'm sympathetic to the "accumulating thousands of small hits to performance" argument, but the question is if this has any measurable impact. Our general philosophy is to prefer readability, unless the performance impact matters.  This is a slippery slope.

In this specific case, I'm not sure it's less readable. But this one seems worse:

```patch
- .setDescription(format("cursor.get%s(%d)", type, field))
+ .setDescription("cursor.get" + type + "(" + field + ")")
```

Hopefully [String Templates](https://openjdk.org/jeps/430) will allow both readability and performance.

Reply:
@electrum - Fair point on the other change, I've pushed a new revision that removes that particular change in `CursorProcessorCompiler`.